Upgrade kubernetes

*Title*: Upgrade kubernetes

*Description*:
As in #766, we are currently "beta" client libraries within the kubernetes versions `kubernetes>=18.20.0,<=21.7.0` that we are using, but in the latest 22.6.0, this beta library is no longer available. Research into upgrading the kubernetes python libraries, and using the correct and compatible kubernetes clients, or remove them if we no longer use or need them.

 *Relevant Links*
#766
https://github.com/kubernetes-client/python/blob/release-22.0/CHANGELOG.md

*To Reproduce*:
Change kubernetes in setup.py to `kubernetes>=18.20.0`

*Logs*:
Before fixing the version to `kubernetes>=18.20.0,<=21.7.0`:
```
cartography/intel/kubernetes/util.py:8: in <module>
    from kubernetes.client import NetworkingV1beta1Api
E   ImportError: cannot import name 'NetworkingV1beta1Api' from 'kubernetes.client' (/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/kubernetes/client/__init__.py)
```

--- a/cartography/intel/kubernetes/util.py
+++ b/cartography/intel/kubernetes/util.py
@@ -5,7 +5,7 @@ from typing import Union
 from kubernetes import config
 from kubernetes.client import ApiClient
 from kubernetes.client import CoreV1Api
-from kubernetes.client import NetworkingV1beta1Api
+from kubernetes.client import NetworkingV1Api
 
 
 class KubernetesContextNotFound(Exception):
@@ -20,7 +20,7 @@ class K8CoreApiClient(CoreV1Api):
         super().__init__(api_client=api_client)
 
 
-class K8NetworkingApiClient(NetworkingV1beta1Api):
+class K8NetworkingApiClient(NetworkingV1Api):
     def __init__(self, name: str, api_client: ApiClient = None) -> None:
         self.name = name
         if not api_client:

--- a/setup.py
+++ b/setup.py
@@ -53,7 +53,7 @@ setup(
         "azure-mgmt-storage>=16.0.0",
         "azure-mgmt-sql<=1.0.0",
         "azure-identity>=1.5.0",
-        "kubernetes>=18.20.0,<=21.7.0",
+        "kubernetes>=22.6.0",
         "pdpyras>=4.3.0",
         "crowdstrike-falconpy>=0.5.1",
     ],

